pkg/policy/defaultRules_test.go

```diff
@@ -0,0 +1,170 @@
+
+	assert.Nil(t, validationError)
+}
+
+func TestDefaultRulesHasUniqueNamesInRules(t *testing.T) {
+	defaultRulesMap, conversionToMapError := convertYamlFileToMap(defaultRulesFileContent)
+	assert.Nil(t, conversionToMapError)
+
+	uniquenessValidationError := validateUniqueStringValuesInRulesForProperty("UniqueName", defaultRulesMap.Rules)
+	assert.Nil(t, uniquenessValidationError)
+}
+
+func TestDefaultRulesHasUniqueIDsInRules(t *testing.T) {
+	defaultRulesMap, conversionToMapError := convertYamlFileToMap(defaultRulesFileContent)
+	assert.Nil(t, conversionToMapError)
+
+	uniquenessValidationError := validateUniqueIntValuesInRulesForProperty("ID", defaultRulesMap.Rules)
+
+	assert.Nil(t, uniquenessValidationError)
+}
+
+func getFileFromPath(path string) (string, error) {
+	fileReader := fileReader.CreateFileReader(nil)
+	fileContent, readFileError := fileReader.ReadFileContent(path)
+
+	if readFileError != nil {
+		return "", readFileError
+	}
+
+	return fileContent, nil
+}
+
+func validateYamlUsingJSONSchema(yamlFilePath string, schema string) error {
+	fileContent, _ := getFileFromPath(yamlFilePath)
+	jsonSchemaValidator := jsonSchemaValidator.New()
+	schemaValidationResult, schemaValidationError := jsonSchemaValidator.ValidateYamlSchema(schema, fileContent)
+
+	if schemaValidationError != nil {
+		panic(errors.New("error occurred while validating yaml file against schema"))
+	}
+
+	if schemaValidationResult != nil {
+		validationErrors := fmt.Errorf("Received validation errors for %s:\n", yamlFilePath)
+
+		for _, validationError := range schemaValidationResult {
+			validationErrors = fmt.Errorf("%s\n%s", validationErrors, validationError.Error)
+		}
+
+		return validationErrors
+	}
+
+	return nil
+}
+
+func convertYamlFileToMap(yamlFileContent string) (DefaultRules, error) {
+	yamlFileContentRawJSON, yamlParseError := yaml.YAMLToJSON([]byte(yamlFileContent))
+
+	if yamlParseError != nil {
+		return DefaultRules{}, yamlParseError
+	}
+
+	var yamlFileContentJSON DefaultRules
+	jsonUnmarshallingError := json.Unmarshal(yamlFileContentRawJSON, &yamlFileContentJSON)
+
+	if jsonUnmarshallingError != nil {
+		return DefaultRules{}, jsonUnmarshallingError
+	}
+
+	return yamlFileContentJSON, nil
+}
+
+func validateUniqueStringValuesInRulesForProperty(propertyName string, rules []DefaultRuleDefinition) error {
+	propertyValuesExistenceMap := make(map[string]bool)
+
+	for _, item := range rules {
+		propertyValue := getStringValueOfRuleProperty(item, propertyName)
+
+		if propertyValuesExistenceMap[propertyValue] {
+			return fmt.Errorf("property %s has duplicate value %s", propertyName, propertyValue)
+		}
+
+		propertyValuesExistenceMap[propertyValue] = true
+	}
+
+	return nil
+}
+
+func getStringValueOfRuleProperty(rule DefaultRuleDefinition, propertyName string) string {
+	itemValue := reflect.ValueOf(rule)
+	propertyValue := reflect.Indirect(itemValue).FieldByName(propertyName).String()
```

Comment:
I really appreciate that you tried to do this generic but I don't see the point here, we just need to make sure item.UniqueName is really unique. I think this function is redundant(+ we don't need to install a new pkg for that). See my suggestion in the father function that uses this.

Reply:
Fair points.

Thought that it would be nice to have for future scenarios where we would want to prevent more duplications, it's premature though.